sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,164 @@
+		return 'Passkey';
+	}
+
+	/**
+	 * Creates an instance of Passkey signer. It's expected to call the static `getPasskeyInstance` method to create an instance.
+	 * For example:
+	 * ```
+	 * const signer = await AwsKmsSigner.getPasskeyInstance();
+	 * ```
+	 */
+	constructor(publicKey: Uint8Array) {
+		super();
+		this.publicKey = publicKey;
+	}
+
+	/**
+	 * Creates an instance of Passkey signer invoking the passkey from navigator.
+	 */
+	static async getPasskeyInstance(): Promise<PasskeyKeypair> {
+		// create a passkey secp256r1 key for the relying party.
+		const credential = (await navigator.credentials.create({
+			publicKey: {
+				challenge: new TextEncoder().encode('Create passkey wallet on Sui'),
+				rp: {
+					name: 'Passkey Wallet on Sui',
+				},
+				user: {
+					id: randomBytes(10),
+					name: 'sui-wallet-user',
+					displayName: 'Sui Wallet User',
+				},
+				pubKeyCredParams: [{ alg: -7, type: 'public-key' }], // -7 is ES256
+				authenticatorSelection: {
+					authenticatorAttachment: 'cross-platform',
+					residentKey: 'required',
+					requireResidentKey: true,
+					userVerification: 'required',
+				},
+				timeout: 60000,
+			},
+		})) as RegistrationCredential;
+
+		if (!credential.response.getPublicKey) {
+			throw new Error('Invalid credential createresponse');
+		} else {
+			const derSPKI = credential.response.getPublicKey()!;
+			const pubkeyUncompressed = parseDerSPKI(new Uint8Array(derSPKI));
+			const pubkey = secp256r1.ProjectivePoint.fromHex(pubkeyUncompressed);
+			const pubkeyCompressed = pubkey.toRawBytes(true);
+			return new PasskeyKeypair(pubkeyCompressed);
+		}
+	}
+
+	/**
+	 * Return the public key for this passkey.
+	 */
+	getPublicKey(): PublicKey {
+		return new PasskeyPublicKey(this.publicKey);
+	}
+
+	/**
+	 * Return the signature for the provided data (i.e. blake2b(intent_message)).
+	 * This is sent to passkey as the challenge field.
+	 */
+	async sign(data: Uint8Array) {
+		// sents the passkey to sign over challenge as the data.
+		const credential = (await navigator.credentials.get({
+			publicKey: {
+				challenge: data,
+				userVerification: 'required',
+				timeout: 60000,
+			},
+		})) as AuthenticationCredential;
+
+		// parse authenticatorData (as bytes), clientDataJSON (decoded as string).
+		const authenticatorData = new Uint8Array(credential.response.authenticatorData);
+		const clientDataJSON = new Uint8Array(credential.response.clientDataJSON); // response.clientDataJSON is already UTF-8 encoded JSON
+		const decoder = new TextDecoder();
+		const clientDataJSONString: string = decoder.decode(clientDataJSON);
+
+		// parse the signature from DER format, normalize and convert to compressed format (33 bytes).
+		const sig = secp256r1.Signature.fromDER(new Uint8Array(credential.response.signature));
+		let normalized = sig.normalizeS().toCompactRawBytes();
+		const compressedPubkey = secp256r1.ProjectivePoint.fromHex(toHex(this.publicKey)).toRawBytes(
```

Comment:
Is this.publicKey already in the right form?  It is already a 33 bytes UInt8Array as far as I can tell.  I would expect that the compressedPubKey in the signature would match the rawBytes from the pubkey instance, if it isn't already in the same format, why are we using different representations?

(sorry if this doesn't make sense, just trying to understand whats happening here)